document_page_portal/security/document_page_portal_security.xml

```diff
@@ -0,0 +1,25 @@
+<?xml version="1.0" encoding="utf-8"?>
+ <odoo>
+    <data noupdate="1">
+
+         <record id="knowledge_portal_document_page_rule" model="ir.rule">
+            <field name="name">Documents: portal users: portal or following</field>
+            <field name="model_id" ref="document_page.model_document_page"/>
+            <field name="domain_force">
+                [
+                '|',
+                    '&amp;',
+                        ('type', '=', 'content'),
```

Comment:
Just a question, just to simplify the query, you are saying: type != 'content' OR is_public or you are follower, isn't it? So, we can write it as:
```
[
'|', '|', ('type', '!=', 'content'), ('is_public', '=', True),
('message_partner_ids', 'child_of', [user.partner_id.commercial_partner_id.id])
]
```

Reply:
@etobella I caught a problem yesterday in the permissions with an internal user without access to the module. I'm going to work on this later this week and I'm already testing with the query suggested by you.

Reply:
@etobella I changed the record rule as you suggested. However, to work properly with internal users and without access to document_page I needed to make some adjustments to the other rules. I don't know if it was the best way and if you have any suggestions I would appreciate it.

cc @ivantodorovich @mileo